app/src/main/java/com/example/fitjournal/home/presentation/components/card/CalisthenicsCard.kt

```diff
@@ -40,10 +39,17 @@ fun CalisthenicsCard(
                     )
             )
             Spacer(modifier = Modifier.height(Spacing.spacing8))
-            MostRecentWorkoutSession()
-            MostRecentSet(
-                reps = reps,
-                time = time,
+            if (calisthenicsUi.sets != null && calisthenicsUi.reps != null) {
+                MostRecentWorkoutSessionTitle()
+            } else {
```

Comment:
I left a comment about moving null checks to an extension function inside data class, we can do that here